sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -861,27 +924,58 @@ def get_job_logs(
                                 break
 
                             # Print logs to the StdOut
-                            print(f"[Pod {pods[index]}]: {logline}")
+                            print(f"[Pod {pods[index].metadata.name}]: {logline}")
                             # Add logs to the results dict.
-                            if pods[index] not in logs_dict:
-                                logs_dict[pods[index]] = logline
+                            if pods[index].metadata.name not in logs_dict:
+                                logs_dict[pods[index].metadata.name] = logline
                             else:
-                                logs_dict[pods[index]] += logline
+                                logs_dict[pods[index].metadata.name] += logline
                         except queue.Empty:
                             break
+                if all(finished):
+                    break
         elif pods:
             for pod in pods:
-                try:
-                    pod_logs = self.core_api.read_namespaced_pod_log(
-                        pod,
-                        namespace,
-                        container=constants.JOB_PARAMETERS[job_kind]["container"],
-                    )
-                    logs_dict[pod] = pod_logs
-                except Exception:
-                    raise RuntimeError(f"Failed to read logs for pod {namespace}/{pod}")
-
-        return logs_dict
+                if (
+                    pod.status is not None
+                    and pod.status.phase != constants.POD_PHASE_PENDING
+                ):
+                    try:
+                        pod_logs = self.core_api.read_namespaced_pod_log(
+                            name=pod.metadata.name,
+                            namespace=namespace,
+                            container=constants.JOB_PARAMETERS[job_kind]["container"],
+                        )
+                        logs_dict[pod.metadata.name] = pod_logs
+                    except Exception:
+                        raise RuntimeError(
+                            f"Failed to read logs for pod {namespace}/{pod.metadata.name}"
+                        )
+        # If verbose is set, return Kubernetes events for Job and pods.
+        if verbose:
+            job = self.get_job(name=name, namespace=namespace)
+            events = self.core_api.list_namespaced_event(namespace=namespace)
+
+            # Get events for the Job and Job's pods.
+            for event in events.items:
```

Comment:
How do we distinguish between job and pod events? Should we add a prefix?

Reply:
Right now, users can use names to get events for Job or Pods.
E.g. for PyTorchJob with name `train-mnist` and 2 workers:
```
{"train-mnist": "Events", "train-mnist-worker-0": "Events", "train-mnist-worker-1": "Events"}
```
Users can understand the all pod names by running `TrainingClient().get_pod_names()`

Do you think we should add the prefix that identify type of the object ?
e.g.
{"pytorchjob-train-mnist": "Events", "pod-train-mnist-worker-0": "Events", "pod-train-mnist-worker-1": "Events"}